sklearn/callback/_callback_context.py

```diff
@@ -207,6 +197,28 @@ def _from_parent(cls, parent_context, *, task_name, task_id, max_subtasks=None):
 
         return new_ctx
 
+    def set_task_info(self, task_name, task_id, max_subtasks=None):
```

Comment:
We can have a default `task_name="fit"` and `task_id=0` to be able to just set `max_subtasks`